app/controllers/reports/regions_controller.rb

```diff
@@ -37,10 +37,21 @@ def show
     @with_ltfu = with_ltfu?
 
     @children = @region.reportable_children
+
+    repository = Reports::Repository.new(@children, periods: @period, with_exclusions: report_with_exclusions?)
     @children_data = @children.map { |child|
-      Reports::RegionService.new(region: child,
-                                 period: @period,
-                                 with_exclusions: report_with_exclusions?).call
+      fetcher = repository.for_region_and_period(child, @period)
```

Comment:
Not sure about this interface yet, but it makes it a bit nicer to grab the one specific datapoint we need for a region and period.

Reply:
Is the alternative here to do `repository[child][@period]`? I think it would be simpler to just use hash lookup semantics if possible, makes it easier to trace things from views to the data.

Reply:
Yup, the fetcher piece may be overkill.  I have a simpler approach in mind for this that I'll try.

Reply:
I think there are a lot of options for how we can go about accessing the data. The approach Prabhanshu mentions is very straight-forward and avoids the additional layer of logic. The combination of having a method that returns an object that references and acts upon the called object, along with the use of method missing, definitely adds some cognitive overhead. And it's not clear that we're getting additional value out of it yet.